src/dotnet/ToolPackage/LocalToolsResolverCache.cs

```diff
@@ -0,0 +1,228 @@
+        }
+
+        public void Save(
+            CommandSettingsListId commandSettingsListId,
+            IReadOnlyList<CommandSettings> commandSettingsList,
+            DirectoryPath nuGetGlobalPackagesFolder)
+        {
+            EnsureFileStorageExists();
+
+            CacheRow cacheRow =
+                ConvertToCacheRow(commandSettingsListId, commandSettingsList, nuGetGlobalPackagesFolder);
+
+            string packageCacheFile = GetCacheFile(commandSettingsListId.PackageId);
+
+            if (_fileSystem.File.Exists(packageCacheFile))
+            {
+                var cacheTable = GetCacheTable(packageCacheFile);
+
+                if (!TryGetMatchingCommandSettingsList(
+                    commandSettingsListId,
+                    nuGetGlobalPackagesFolder,
+                    cacheTable,
+                    out IReadOnlyList<CommandSettings> _))
+                {
+                    _fileSystem.File.WriteAllText(
+                        packageCacheFile,
+                        JsonConvert.SerializeObject(
+                            cacheTable.Concat(new[] {cacheRow}).ToArray()));
+                }
+            }
+            else
+            {
+                _fileSystem.File.WriteAllText(
+                    packageCacheFile,
+                    JsonConvert.SerializeObject(new[] {cacheRow}));
+            }
+        }
+
+        public IReadOnlyList<CommandSettings> Load(
+            CommandSettingsListId commandSettingsListId,
+            DirectoryPath nuGetGlobalPackagesFolder)
+        {
+            string packageCacheFile = GetCacheFile(commandSettingsListId.PackageId);
+            if (_fileSystem.File.Exists(packageCacheFile))
+            {
+                if (TryGetMatchingCommandSettingsList(
+                    commandSettingsListId,
+                    nuGetGlobalPackagesFolder,
+                    GetCacheTable(packageCacheFile),
+                    out IReadOnlyList<CommandSettings> commandSettingsList))
+                {
+                    return commandSettingsList;
+                }
+            }
+
+            return Array.Empty<CommandSettings>();
+        }
+
+        private CacheRow[] GetCacheTable(string packageCacheFile)
+        {
+            CacheRow[] cacheTable = Array.Empty<CacheRow>();
+
+            try
+            {
+                cacheTable =
+                    JsonConvert.DeserializeObject<CacheRow[]>(_fileSystem.File.ReadAllText(packageCacheFile));
+            }
+            catch (JsonReaderException)
+            {
+                // if file is corrupted, treat it as empty since it is not the source of truth
+            }
+
+            return cacheTable;
+        }
+
+        public bool TryLoadHighestVersion(
+            CommandSettingsListIdVersionRange query,
+            DirectoryPath nuGetGlobalPackagesFolder,
+            out IReadOnlyList<CommandSettings> commandSettingsList)
```

Comment:
Return single object instead of the `out` list.

Reply:
Fixed as suggested